core/src/test/java/io/grpc/internal/ManagedChannelImplTest.java

```diff
@@ -2148,13 +2237,22 @@ void allResolved() {
       Listener listener;
       boolean shutdown;
       int refreshCalled;
+      Status error;
+
+      FakeNameResolver(Status error) {
+        this.error = error;
+      }
 
       @Override public String getServiceAuthority() {
         return expectedUri.getAuthority();
       }
 
       @Override public void start(final Listener listener) {
         this.listener = listener;
+        if (error != null) {
```

Comment:
Can we consolidate this with the error simulation in refresh() and put it in resolved()?

Reply:
Done. Also made a larger scale refactor to switch to a builder for the FakeNameResolverFactory, as it was confusing sorting through the four different constructors and their different default values.